components/BaseContentController.php

```diff
@@ -336,4 +339,191 @@ protected function prepareRequestParams(array $requestParams, string $formName,
+
+        if (!$this->updatePinned($activeRecord, $data['metadata'])) {
+            return false;
+        }
+
+        if (!$this->updateLockedComments($activeRecord, $data['metadata'])) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function updatePinned(ContentActiveRecord $activeRecord, array $data): bool
+    {
+        if (!isset($data['pinned'])) {
+            return true;
+        }
+
+        if (!in_array($data['pinned'], [0, 1])) {
+            $activeRecord->addError('pinned', 'Wrong field "pinned" value!');
+            return false;
+        }
+
+        if ($data['pinned']) {
+            if (!$activeRecord->content->canPin()) {
+                $activeRecord->addError('archived', 'You cannot pin the Content!');
+                return false;
+            }
+            $activeRecord->content->pin();
+        } else {
+            $activeRecord->content->unpin();
+        }
+
+        return true;
+    }
+
+    protected function updateArchived(ContentActiveRecord $activeRecord, array $data): bool
+    {
+        if (!isset($data['archived'])) {
+            return true;
+        }
+
+        if (!in_array($data['archived'], [0, 1])) {
+            $activeRecord->addError('archived', 'Wrong field "archived" value!');
+            return false;
+        }
+
+        if (!$activeRecord->content->canArchive()) {
+            $activeRecord->addError('archived', 'You cannot archive the Content!');
+            return false;
+        }
+
+        if ($data['archived']) {
+            $activeRecord->content->archive();
+        } else {
+            $activeRecord->content->unarchive();
+        }
+
+        return true;
+    }
+
+    protected function updateVisibility(ContentActiveRecord $activeRecord, array $data): bool
+    {
+        if (!isset($data['visibility'])) {
+            return true;
+        }
+
+        if (!in_array($data['visibility'], [Content::VISIBILITY_PRIVATE, Content::VISIBILITY_PUBLIC, Content::VISIBILITY_OWNER])) {
+            $activeRecord->addError('archived', 'Wrong field "visibility" value!');
+            return false;
+        }
+
+        $activeRecord->content->visibility = $data['visibility'];
+        return $activeRecord->content->save();
+    }
+
+    protected function updateLockedComments(ContentActiveRecord $activeRecord, array $data): bool
+    {
+        if (!isset($data['locked_comments'])) {
+            return true;
+        }
+
+        if (!in_array($data['locked_comments'], [0, 1])) {
+            $activeRecord->addError('locked_comments', 'Wrong field "locked_comments" value!');
+            return false;
+        }
+
+        if (!$activeRecord->content->canLockComments()) {
+            $activeRecord->addError('locked_comments', 'You cannot lock comments of the Content!');
+            return false;
+        }
+
+        $activeRecord->content->locked_comments = $data['locked_comments'];
+        return $activeRecord->content->save();
+    }
```

Comment:
#88 related?

Reply:
@donni106 Yes, partly related, I answered there https://github.com/humhub/rest/issues/88#issuecomment-1123562948, thank you for the report!